pay-api/src/pay_api/services/direct_pay_service.py

```diff
@@ -12,28 +12,89 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Service to manage Direct Pay PAYBC Payments."""
+import json
+import secrets
+import urllib.parse
 
-from datetime import datetime
+from datetime import datetime, date
 from typing import Any, Dict
 
 from flask import current_app
 
 from pay_api.services.base_payment_system import PaymentSystemService
+from pay_api.services.hashing import HashingService
 from pay_api.services.invoice import Invoice
 from pay_api.services.invoice_reference import InvoiceReference
 from pay_api.services.payment_account import PaymentAccount
 from pay_api.utils.enums import PaymentSystem, PaymentMethod
+from pay_api.models.distribution_code import DistributionCode as DistributionCodeModel
+from pay_api.models.payment_line_item import PaymentLineItem as PaymentLineItemModel
 from .oauth_service import OAuthService
 from .payment_line_item import PaymentLineItem
 
+PAYBC_DATE_FORMAT = '%Y-%M-%d'
+PAYBC_REVENUE_SEPARATOR = '|'
+DECIMAL_PRECISION = '.2f'
+
 
 class DirectPayService(PaymentSystemService, OAuthService):
     """Service to manage internal payment."""
 
     def get_payment_system_url(self, invoice: Invoice, inv_ref: InvoiceReference, return_url: str):
         """Return the payment system url."""
-        # TODO return the proper url .Have to be addressed in the next task
-        return None
+        today = date.today().strftime(PAYBC_DATE_FORMAT)
+
+        url_params = {'trnDate': today,
+                      'pbcRefNumber': current_app.config.get('PAYBC_DIRECT_PAY_REF_NUMBER'),
+                      'glDate': today,
+                      'description': 'DIRECT SALE',
+                      'trnNumber': inv_ref.reference_number,
```

Comment:
trnNumber should be the invoice_id, we need to make sure there is only 1 record in PayBC if it's direct pay and trnNumber is unique, and doesn't allow duplicate payment for same trnNumber. For PayBC invoiced CC flow, there can be multiple references as delete payment exists.

Reply:
I changed the txnNumber .Now it is 'trnNumber': invoice.id,